### PR TITLE
SNS - Temperature sensor REPL commands

### PR DESCRIPTION
**Pull request by TomLonergan03**
This works!

### CODE REVIEW

**Review by davidbeechey — APPROVED** [review]

File: lib/sensors/temperature.hpp [hunk 1]
```diff
@@ -11,16 +11,14 @@
 
 namespace hyped::sensors {
 
-// Two possible addresses for the temperature sensor
-constexpr std::uint8_t kDefaultTemperatureAddress     = 0x38;
-constexpr std::uint8_t kAlternativeTemperatureAddress = 0x3F;
+enum class temperatureAddress { k7F = 0x7F, k38 = 0x38, k3C = 0x3C, k3E = 0x3E };
```
Comment (licornes-fluos) [inline lib/sensors/temperature.hpp:14]: According to the datasheet, the addresses are 0x70/78/7C/7E

**Review by licornes-fluos — CHANGES_REQUESTED** [review]

File: lib/debug/commands/temperature_commands.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  };
+  const auto i2c              = std::move(*optional_i2c);
+  const auto optional_address = config["address"].value<std::uint8_t>();
+  if (!optional_address) {
+    logger.log(core::LogLevel::kFatal, "Invalid address");
+    return core::Result::kFailure;
+  }
+  const auto address                     = std::move(*optional_address);
+  const auto optional_temperature_sensor = sensors::Temperature::create(
+    logger, i2c, 0, static_cast<sensors::temperatureAddress>(address));
```
Comment (licornes-fluos) [inline lib/debug/commands/temperature_commands.cpp:30]: Shouldn't this be bus 2, as defined in the debug.toml?

File: lib/sensors/temperature.hpp (hunk 1 above)
Comment (TomLonergan03) [reply lib/sensors/temperature.hpp:14]: uh where are you seeing that? the sensor definitely shows up at 0x38

File: lib/debug/commands/temperature_commands.cpp (hunk 2 above)
Comment (TomLonergan03) [reply lib/debug/commands/temperature_commands.cpp:30]: that's the mux channel. we'll merge this, then when the accelerometer PR gets merged there will be a cleanup

**Review by licornes-fluos — APPROVED** [review]

**Pull request merged**